build/media_source/system/js/inline-editing.es6.js

```diff
@@ -0,0 +1,163 @@
+    cancelButton.type = 'button';
+    cancelButton.classList.add('btn-danger');
+    Object.assign(cancelButton.style, {
+      float: 'left',
+      margin: '2px',
+    });
+    cancelButton.appendChild(cancelButtonIcon);
+
+    // Loading icon
+    loader.src = icon;
+    loader.width = 30;
+    loader.height = 30;
+    Object.assign(loader.style, {
+      position: 'absolute',
+      top: 0,
+      bottom: 0,
+      left: 0,
+      right: 0,
+      margin: 'auto',
+    });
+    loader.classList.add('d-none');
+
+    // Wrap textArea and loader inside a div
+    wrap.style.position = 'relative';
+    wrap.style.display = 'inline-block';
+    wrap.appendChild(cancelButton);
+    wrap.appendChild(textArea);
+    wrap.appendChild(loader);
+
+    return {
+      wrap,
+      textArea,
+      loader,
+      cancelButton,
+    };
+  };
+
+  const Error = (element) => {
+    element.classList.add('highlight-error');
+    setTimeout(() => {
+      element.classList.remove('highlight-error');
+    }, 2000);
+  };
+
+  // Handle custom fields
+  const FieldHelper = ({ itemId, fieldId, _this }) => {
+    const oldContent = _this.innerText;
+
+    const wrapAndTextArea = GetTextAreaWrapper(oldContent);
+    if (!wrapAndTextArea) {
+      return;
+    }
+    const {
+      wrap,
+      textArea,
+      loader,
+      cancelButton,
+    } = wrapAndTextArea;
+
+    // Add wrap to the DOM
+    _this.parentNode.insertBefore(wrap, _this.nextSibling);
+
+    // Hide custom field value and show a text area
+    _this.classList.add('d-none');
+    textArea.focus();
+
+    let cancel = false;
+    // Restore original element on Cancel
+    cancelButton.addEventListener('click', () => {
+      cancel = true;
+      wrap.remove();
+      _this.classList.remove('d-none');
+    });
+
+    // Send Ajax request and update front-end when user focuses out of textArea
+    textArea.addEventListener('focusout', () => {
+      const AjaxCall = () => {
+        const newValue = textArea.value;
+        const url = '?option=com_fields&task=Field.saveField&format=json';
+        const data = `field_id=${fieldId}&item_id=${itemId}&value=${newValue}&${Joomla.getOptions('csrf.token', '')}=1`;
+
+        textArea.disabled = true;
+        loader.classList.remove('d-none');
+
+        const AjaxEnd = () => {
+          wrap.remove();
+          _this.classList.remove('d-none');
+        };
+
+        Joomla.request({
+          url,
+          method: 'POST',
+          data,
+          onSuccess: (response) => {
+            const responseJson = JSON.parse(response);
+
+            if (responseJson.data.saved === true) {
+              _this.innerHTML = newValue;
+            } else {
+              Error(_this);
+            }
+
+            AjaxEnd();
+          },
+          onError: () => {
+            Error(_this);
+            AjaxEnd();
+          },
+        });
+      };
+
+      setTimeout(() => {
+        if (!document.hasFocus() || cancel) {
```

Comment:
Normally, `document.hasFocus()` always returns `true`. So, I have put a delay of 200ms here. But this might not work all the time. What condition should I use so that the `focusout` event fires only if window is still in focus and text-area focus has been lost.

Reply:
@rs4231199 The document always has focus because that is the page that is open. You will want to add an event listener on the field itself. Let's say the field has an ID of `article-title` you would do this:
```js
document.getElementById('article-title').addEventListener('focusout', event => {console.log(event.target)})
```
Whenever the user leaves that field, this trigger will be called.

Reply:
If I add eventListener to `text-area` then It save the content when user changes window.
![3036673a-2b9b-4753-a223-5b2c8313df0c](https://user-images.githubusercontent.com/43215292/123984469-46e23c00-d9e2-11eb-8550-289c8ee83b2e.gif)

How do I check if the window/tab is still in focus? Otherwise It will save the content whenever user leaves the window/tab.

Reply:
> If I add eventListener to text-area then It saves the content when user changes window.

Of course, that is moving focus and you have a trigger for `focusout`. I don't believe you can have a listener for what you want. The only option then is to add a save button I would say. You could check if there is a trigger that will do what you want but I don't see how.

Reply:
Okay. I will add a save button for now.